src/tests/ftest/util/general_utils.py

```diff
@@ -666,6 +666,54 @@ def convert_list(value, separator=","):
     return separator.join([str(item) for item in value])
 
 
+def dump_engines_stacks(hosts, verbose=True, timeout=60, added_filter=None):
+    """Signal the engines on each hosts tog their ULT stacks dump.
+
+    Args:
+        hosts (list): hosts on which to signal the engines
+        verbose (bool, optional): display command output. Defaults to True.
+        timeout (int, optional): command timeout in seconds. Defaults to 60
+            seconds.
+        added_filter (str, optional): negative filter to better identify
+            engines.
+
+
+    Returns:
+        dict: a dictionary of return codes keys and accompanying NodeSet
+            values indicating which hosts yielded the return code.
+            Return code keys:
+                0   No engine matched the criteria / No engine signaled.
+                1   One or more engine matched the criteria and a signal was
+                    sent.
+
+    """
+    result = {}
+    log = getLogger()
+    log.info("Dumping ULT stacks of engines on %s", hosts)
+
+    if added_filter:
+        ps_cmd = "/usr/bin/ps xa | grep daos_engine | grep -vE {}".format(
+            added_filter)
+    else:
+        ps_cmd = "/usr/bin/pgrep --list-full daos_engine"
+
+    if hosts is not None:
+        commands = [
+            "rc=0",
+            "if " + ps_cmd,
+            "then rc=1",
+            "sudo pkill --signal USR2 daos_engine",
+            # leave time for ABT info/stacks dump vs xstream/pool/ULT number
+		"sleep 30",
```

Comment:
Is there a more deterministic way to know that the ULT stacks have been dumped?  We strongly discourage using sleeps in our test code.  In addition this will most likely cause some test timeout issues with the way avocado handles the test timeout.  We currently allow an additional 60 seconds for `tearDown()` to run if the test times out (https://github.com/daos-stack/daos/blob/master/src/tests/ftest/scripts/main.sh#L81, https://github.com/daos-stack/daos/blob/master/src/tests/ftest/scripts/main.sh#L174, https://github.com/daos-stack/daos/blob/master/src/tests/ftest/scripts/main.sh#L183) so we may need to increase this for the worse case scenario.  There may also be some tests that manage to complete their `setUp()`, `test_*()` , and `tearDown()` method executions with their current `self.timeout` value, (even if it fails) but adding a fixed 30 second delay to the `tearDown()` execution time (when there is a failure) may cause that test to also timeout in `tearDown()` - something we'd like to avoid to ensure a failing test does not impact the next test.

Reply:
I understand this hard timer of 30s is clearly too basic, but at this time, there is no way to detect when ULTs stacks dump is done or still running. I am presently working with Argobots team to give us the possibility to detect it (see ABT issue [#372](https://github.com/pmodels/argobots/issues/372)), and we may fix this when this will be available..